api/pkg/gcs/gcs.go

```diff
@@ -0,0 +1,59 @@
+package gcs
+
+import (
+	"context"
+	"strings"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/iterator"
+)
+
+type gcsClient struct {
+	Api    *storage.Client
+	Config Config
+}
+type Config struct {
+	Ctx context.Context
+}
+type GcsService interface {
+	DeleteArtifact(url string) error
+}
+
+func NewGcsClient(api *storage.Client, cfg Config) *gcsClient {
+	return &gcsClient{
+		Api:    api,
+		Config: cfg,
+	}
+}
+
+func (gc *gcsClient) DeleteArtifact(url string) error {
+	// Get bucket name and gcsPrefix
+	gcsBucket, gcsLocation := gc.RemoveAndSplit(url, "/")
+
+	// Sets the name for the bucket.
+	bucket := gc.Api.Bucket(gcsBucket)
+
+	it := bucket.Objects(gc.Config.Ctx, &storage.Query{
+		Prefix: gcsLocation,
+	})
+	for {
+		attrs, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return err
+		}
+		if err := bucket.Object(attrs.Name).Delete(gc.Config.Ctx); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (gc *gcsClient) RemoveAndSplit(str, delimiter string) (string, string) {
```

Comment:
Can this method be named something like `getGcsBucketAndLocation` since we're expecting it only perform one purpose: to retrieve those values from a url. We aren't using it elsewhere and it can't be extended for any other use so giving it a specific name would be more useful. You can also remove the `delimiter` argument in this case. (Note also the lower-case function name since we aren't going to call this function from outside the `gcs` package). 

Also, you might want to insert additional inline comments to show an example of how splitting a URL would return the GCS bucket and location values.